src/aitsi/m3spin/pkb/impl/ParentImpl.java

```diff
@@ -0,0 +1,49 @@
+package aitsi.m3spin.pkb.impl;
+
+import aitsi.m3spin.commons.interfaces.TNode;
+import aitsi.m3spin.pkb.interfaces.Parent;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ParentImpl implements Parent {
+
+    @Override
+    public List<TNode> getParentedBy(TNode p) {
+
+        TNode n = p;
```

Comment:
Po co to przypisanie? Skoro argument `p` nie jest poprzedzony słówkiem kluczowym `final`, to można go modyfikować

Reply:
Wszystko można, tylko później w kodzie cudownie wygląda childList.add(parent);